lib/fog/google/models/storage_json/files.rb

```diff
@@ -0,0 +1,93 @@
+require "fog/core/collection"
+require "fog/google/models/storage_json/file"
+
+module Fog
+  module Storage
+    class GoogleJSON
+      class Files < Fog::Collection
+        extend Fog::Deprecation
+        deprecate :get_url, :get_https_url
+
+        attribute :common_prefixes, :aliases => "CommonPrefixes"
+        attribute :delimiter,       :aliases => "Delimiter"
+        attribute :directory
+        # attribute :is_truncated,    :aliases => "IsTruncated"
+        attribute :page_token,      :aliases => %w(pageToken page_token)
+        attribute :max_results,     :aliases => ["MaxKeys", "max-keys"]
+        attribute :prefix,          :aliases => "Prefix"
+
+        model Fog::Storage::GoogleJSON::File
+
+        def all(options = {})
+          requires :directory
+          options = {
+            "delimiter"   => delimiter,
+            "pageToken"   => page_token,
+            "maxResults"  => max_results,
+            "prefix"      => prefix
+          }.merge!(options)
+          options = options.reject { |_key, value| value.nil? || value.to_s.empty? }
+          merge_attributes(options)
+          parent = directory.collection.get(
+            directory.key,
+            options
+          )
+          if parent
+            merge_attributes(parent.files.attributes)
+            load(parent.files.map(&:attributes))
+          end
+        end
+
+        alias_method :each_file_this_page, :each
+        def each
+          if !block_given?
+            self
+          else
+            subset = dup.all
+
+            subset.each_file_this_page { |f| yield f }
+            while subset.is_truncated
+              subset = subset.all(:marker => subset.last.key)
+              subset.each_file_this_page { |f| yield f }
+            end
+
+            self
+          end
+        end
+
+        def get(key, options = {}, &block)
+          requires :directory
+          data = service.get_object(directory.key, key, options, &block)
+          file_data = {}
+          data.headers.each do |key, value|
```

Comment:
'key' is shadowing the outer parameter of 'get'. Suggest replacing key, value with |k,v|